### ISSUE TITLE
LDAP : how to manage users not found anymore by LDAP filter ?

### ISSUE THREAD

**Issue by Yvan-Masson** [owncloud/core#23490]
First, please apologize if this as already been discussed but I could not find it.

### Steps to reproduce
1. Configure ownCloud 8.2 to talk with a LDAP server
2. Connect to ownCloud with a LDAP user so that it is know to ownCloud
3. Modify this user in LDAP so that the LDAP filter in ownCloud does not match this user anymore (account expired / disabled for example).

### Expected behaviour
This should be expected by ownCloud internals, and it should be possible to delete the user and his data in ownCloud with a defined behaviour.

### Actual behaviour
All documents and shares are kept back, in case the LDAP entry is modified again and the ownCloud LDAP filter matches him again -> this is the great part.

But, as explained in #22024, before ownCloud 9, the user is not detected as a "remnant" by ldapUserCleanupInterval, which implies that:
 - the user can't be deleted with "sudo -u www-data php occ user:delete username"
 - there is no way to delete the user and his documents in a proper way (what happens if some documents were shared with others ?)

### Server configuration
**Operating system**:
 Debian 8.3
**Web server:**
Apache
**Database:**
MariaDB
**PHP version:**
5.6.17
**ownCloud version:** (see ownCloud admin page)
8.2.3
**Updated from an older ownCloud or fresh install:**
Fresh install.
**Where did you install ownCloud from:**
deb from http://download.owncloud.org/download/repositories/8.2/Debian_8.0/

**List of activated apps:**
Enabled:
  - activity: 2.1.4
  - calendar: 0.8.2
  - contacts: 0.5.0.0
  - files: 1.2.1
  - files_pdfviewer: 0.7
  - files_sharing: 0.7.0
  - files_texteditor: 2.0
  - files_trashbin: 0.7.0
  - files_versions: 1.1.0
  - firstrunwizard: 1.1
  - gallery: 14.2.0
  - notifications: 0.1.0
  - provisioning_api: 0.3.0
  - templateeditor: 0.1
  - user_ldap: 0.7.1
Disabled:
  - encryption
  - external
  - files_external
  - files_videoviewer
  - user_external

```
**The content of config/config.php:**
{
    "system": {
        "instanceid": "ocednpxippb9",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "owncloud.lama.univ-savoie.fr",
            "owncloud.lama.univ-smb.fr",
            "193.48.120.32"
        ],
        "datadirectory": "\/var\/www\/owncloud\/data",
        "overwrite.cli.url": "http:\/\/owncloud.lama.univ-savoie.fr",
        "dbtype": "mysql",
        "version": "8.2.3.2",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "asset-pipeline.enabled": true,
        "mail_smtpmode": "smtp",
        "mail_smtphost": "localhost",
        "mail_from_address": "owncloud",
        "mail_domain": "lama.univ-savoie.fr",
        "ldapIgnoreNamingRules": false,
        "ldapUserCleanupInterval": 51,
        "loglevel": 0,
        "log_rotate_size": 10485760,
        "skeletondirectory": "\/var\/www\/owncloud\/core\/LAMA-skeleton",
        "filelocking.enabled": "true",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "\/var\/run\/redis\/redis.sock",
            "port": 0,
            "timeout": 0
        }
    }
}
```
**Are you using external storage, if yes which one:** local/smb/sftp/...
No
**Are you using encryption:** yes/no
No
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
LDAP
#### LDAP configuration (delete this part if not used)
```
hasMemberOfFilterSupport      | 1                                                                                                                                                                                |
| hasPagedResultSupport         |                                                                                                                                                                                  |
| homeFolderNamingRule          | attr:cn                                                                                                                                                                          |
| lastJpegPhotoLookup           | 0                                                                                                                                                                                |
| ldapAgentName                 | cn=ldapUser,cn=Users,dc=ad,dc=lama                                                                                                                                               |
| ldapAgentPassword             | ***                                                                                                                                                                              |
| ldapAttributesForGroupSearch  |                                                                                                                                                                                  |
| ldapAttributesForUserSearch   |                                                                                                                                                                                  |
| ldapBackupHost                |                                                                                                                                                                                  |
| ldapBackupPort                |                                                                                                                                                                                  |
| ldapBase                      | dc=ad,dc=lama                                                                                                                                                                    |
| ldapBaseGroups                | dc=ad,dc=lama                                                                                                                                                                    |
| ldapBaseUsers                 | dc=ad,dc=lama                                                                                                                                                                    |
| ldapCacheTTL                  | 600                                                                                                                                                                              |
| ldapConfigurationActive       | 1                                                                                                                                                                                |
| ldapEmailAttribute            | mail                                                                                                                                                                             |
| ldapExperiencedAdmin          | 0                                                                                                                                                                                |
| ldapExpertUUIDGroupAttr       |                                                                                                                                                                                  |
| ldapExpertUUIDUserAttr        |                                                                                                                                                                                  |
| ldapExpertUsernameAttr        |                                                                                                                                                                                  |
| ldapGroupDisplayName          | cn                                                                                                                                                                               |
| ldapGroupFilter               |                                                                                                                                                                                  |
| ldapGroupFilterGroups         |                                                                                                                                                                                  |
| ldapGroupFilterMode           | 0                                                                                                                                                                                |
| ldapGroupFilterObjectclass    |                                                                                                                                                                                  |
| ldapGroupMemberAssocAttr      | uniqueMember                                                                                                                                                                     |
| ldapHost                      | ldaps://lama-dc.local.domain.fr                                                                                                                                             |
| ldapIgnoreNamingRules         |                                                                                                                                                                                  |
| ldapLoginFilter               | (&(&(|(objectclass=person)))(samaccountname=%uid))                                                                                                                               |
| ldapLoginFilterAttributes     |                                                                                                                                                                                  |
| ldapLoginFilterEmail          | 0                                                                                                                                                                                |
| ldapLoginFilterMode           | 1                                                                                                                                                                                |
| ldapLoginFilterUsername       | 1                                                                                                                                                                                |
| ldapNestedGroups              | 0                                                                                                                                                                                |
| ldapOverrideMainServer        | 0                                                                                                                                                                                |
| ldapPagingSize                | 500                                                                                                                                                                              |
| ldapPort                      | 636                                                                                                                                                                              |
| ldapQuotaAttribute            |                                                                                                                                                                                  |
| ldapQuotaDefault              |                                                                                                                                                                                  |
| ldapTLS                       | 0                                                                                                                                                                                |
| ldapUserDisplayName           | displayname                                                                                                                                                                      |
| ldapUserDisplayName2          |                                                                                                                                                                                  |
| ldapUserFilter                | (&(objectCategory=CN=Person,CN=Schema,CN=Configuration,DC=ad,DC=lama)(!(UserAccountControl:1.2.840.113556.1.4.803:=2))(accountExpires>=131031001420000000)(!(accountExpires=0))) |
| ldapUserFilterGroups          |                                                                                                                                                                                  |
| ldapUserFilterMode            | 0                                                                                                                                                                                |
| ldapUserFilterObjectclass     | person                                                                                                                                                                           |
| ldapUuidGroupAttribute        | auto                                                                                                                                                                             |
| ldapUuidUserAttribute         | auto                                                                                                                                                                             |
| turnOffCertCheck              | 0                                                                                                                                                                                |
| useMemberOfToDetectMembership | 1
```

**Comment by PVince81** [comment]
@blizzz @MorrisJobke

**Comment by blizzz** [comment]
please see https://doc.owncloud.org/server/8.2/admin_manual/configuration_user/user_auth_ldap_cleanup.html

**Comment by cdamken** [comment]
@Yvan-Masson  Did you fixed changing your LDAP configuration?

**Comment by Yvan-Masson** [comment]
@cdamken I had no time before now to get back on this issue. It will be fortunately fixed for the next for 8.1 and 8.2 maintenance releases, see end of #25338.
